File: app/src/main/java/com/imangazalievm/bubbble/di/ShotDetailsPresenterComponent.java
package com.imangazalievm.bubbble.di;

import com.imangazalievm.bubbble.di.modules.ShotDetailsPresenterModule;
import com.imangazalievm.bubbble.di.scopes.Presenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.ShotDetailsPresenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.ShotsPresenter;

import dagger.Component;

@Presenter
@Component(dependencies = ApplicationComponent.class, modules = ShotDetailsPresenterModule.class)
public interface ShotDetailsPresenterComponent {

    ShotDetailsPresenter getPresenter();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/shotdetails/ShotDetailsPresenterComponent.kt
package com.imangazalievm.bubbble.di.shotdetails

import com.imangazalievm.bubbble.di.global.ApplicationComponent
import com.imangazalievm.bubbble.di.global.scopes.Presenter
import com.imangazalievm.bubbble.presentation.shotdetails.ShotDetailsPresenter
import dagger.Component

@Presenter
@Component(
    dependencies = [ApplicationComponent::class],
    modules = [ShotDetailsPresenterModule::class]
)
interface ShotDetailsPresenterComponent {

    fun getPresenter(): ShotDetailsPresenter

}